.appveyor/build_task.cmd

```diff
@@ -0,0 +1,73 @@
+@echo off
+setlocal enableextensions enabledelayedexpansion
+
+	cd /D %APPVEYOR_BUILD_FOLDER%
+	if %errorlevel% neq 0 exit /b 3
+
+	set STABILITY=staging
+	set DEPS_DIR=%PHP_BUILD_CACHE_BASE_DIR%\deps-%PHP_REL%-%PHP_SDK_VC%-%PHP_SDK_ARCH%
+	rem SDK is cached, deps info is cached as well
+	echo Updating dependencies in %DEPS_DIR%
+	cmd /c phpsdk_deps --update --no-backup --branch %PHP_REL% --stability %STABILITY% --deps %DEPS_DIR%
+	if %errorlevel% neq 0 exit /b 3
+
+	rem Something went wrong, most likely when concurrent builds were to fetch deps
+	rem updates. It might be, that some locking mechanism is needed.
+	if not exist "%DEPS_DIR%" (
+		cmd /c phpsdk_deps --update --force --no-backup --branch %PHP_REL% --stability %STABILITY% --deps %DEPS_DIR%
+	)
+	if %errorlevel% neq 0 exit /b 3
+
+	if "%ZTS_STATE%"=="enable" set ZTS_SHORT=ts
+	if "%ZTS_STATE%"=="disable" set ZTS_SHORT=nts
+	if "%ZTS_STATE%"=="enable" set ZTS_IN_FILENAME=
+	if "%ZTS_STATE%"=="disable" set ZTS_IN_FILENAME=-nts
+	if "%PHP_SDK_ARCH%"=="x86" set ARCH_IN_FILENAME=
+	if "%PHP_SDK_ARCH%"=="x64" set ARCH_IN_FILENAME=-x86_64
+
+	cd /d C:\projects\php-src
+
+	cmd /c buildconf.bat --force
+
+	if %errorlevel% neq 0 exit /b 3
+
+	cmd /c configure.bat --disable-all --with-mp=auto --enable-cli --%ZTS_STATE%-zts --enable-json --with-openssl --enable-mongodb=shared --enable-object-out-dir=%PHP_BUILD_OBJ_DIR% --with-config-file-scan-dir=%APPVEYOR_BUILD_FOLDER%\build\modules.d --with-prefix=%APPVEYOR_BUILD_FOLDER%\build --with-php-build=%DEPS_DIR%
+
+	if %errorlevel% neq 0 exit /b 3
+
+	nmake /NOLOGO
+	if %errorlevel% neq 0 exit /b 3
+
+	nmake install
+
+	if %errorlevel% neq 0 exit /b 3
+
+	mkdir c:\tests_tmp
```

Comment:
I missed that the test suite was being executed here.

I think this really belongs in its own `test_script.cmd` file, which can be run from the `test_script` phase in `appveyor.yml`. https://www.appveyor.com/docs/appveyor-yml/ has an example of this:

```yml
# to run your custom scripts instead of automatic tests
test_script:
  - echo This is my custom test script
```

Reply:
If this poses problems with making the build fail (as you mentioned in https://github.com/mongodb/mongo-php-driver/pull/764#discussion_r170301548), which can't be resolved by [Allow Failing Jobs](https://www.appveyor.com/docs/build-configuration/#allow-failing-jobs), then we can leave this as-is. Please make a PHPC ticket to revisit this later, though.

Reply:
The problem was more that in order to run the tests in a separate script, you'd have to store the output of the build tasks, and basically restore things for every task. Not worth the effort, as this works just fine™. I wouldn't even want to add a PHPC ticket for this.